ctk/common/src/test/kotlin/org/codice/compilance/verification/core/responses/CoreAuthnRequestProtocolVerifierSpec.kt

```diff
@@ -48,17 +52,18 @@ class CoreAuthnRequestProtocolVerifierSpec : StringSpec() {
                     CoreAuthnRequestProtocolVerifier(AuthnRequestBuilder().buildObject(), it)
                             .verifyAuthnContextClassRef()
                 }
+                Report.hasExceptions().shouldBeFalse()
             }
         }
 
         "response with incorrect AuthnContextClassRef fails" {
             forAll(listOf("wrong", "", null)) { authContext ->
                 NodeDecorator(Common.buildDom(response(authContext))).let {
-                    shouldThrow<SAMLComplianceException> {
-                        CoreAuthnRequestProtocolVerifier(AuthnRequestBuilder().buildObject(), it)
-                                .verifyAuthnContextClassRef()
-                    }.message?.shouldContain(SAMLCore_3_3_2_2_1_a.message)
+                    CoreAuthnRequestProtocolVerifier(AuthnRequestBuilder().buildObject(), it)
+                            .verifyAuthnContextClassRef()
                 }
+                Report.getExceptionMessages(CORE_3_3).shouldContain(SAMLCore_3_3_2_2_1_a.message)
+                Report.resetExceptionMap()
```

Comment:
❓ Do you really need this everywhere, since you've added the report listener?

Reply:
I added this here cause it's in a forAll loop so the listener won't clear it